app/models/training_module.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+# == Schema Information
+#
+# Table name: training_slides
+#
+#  id              :bigint(8)        not null, primary key
+#  name            :string(255)
+#  estimated_ttc   :string(255)
+#  wiki_page       :string(255)
+#  slide_slugs     :text(65535)
+#  description     :text(65535)
+#  slug            :string(255)
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+
+require_dependency "#{Rails.root}/lib/training/training_base"
+require_dependency "#{Rails.root}/lib/training_library"
+
+#= Class representing an individual training module
+class TrainingModule < ApplicationRecord
+  attr_accessor :status
+  serialize :slide_slugs, Array
+
+  validates :id, uniqueness: true
+  validates :slug, uniqueness: true
+
+  def self.path_to_yaml
+    "#{base_path}/modules/*.yml"
+  end
+
+  def self.wiki_base_page
+    ENV['training_modules_wiki_page']
+  end
+
+  def self.trim_id_from_filename
+    false
+  end
+
+  def self.load
+    TrainingBase.load(content_class: self)
+  end
+
+  def self.base_path
+    TrainingBase.base_path
+  end
+
+  def self.load_all
+    TrainingBase.load_all
+  end
+
+  def self.flush
+    delete_all
+  end
+
+  # This reloads all the library and module content, but only updates the slides
+  # for the module with the given slug.
+  def self.reload_module(slug:)
```

Comment:
I think this can be simplified, now that TrainingModule is ActiveRecord, instead of long-lived mutable objects in shared memory. We shouldn't ever need to use `.flush`, or `load` without the slug list, except for with TrainingLibrary.

A big part of the benefit of doing this transition to ActiveRecord is to simplify the reloading process and remove unnecessary slow network calls to fetch and refetch Wikipedia content.